File: 3958-minimum-removals-to-balance-array/minimum-removals-to-balance-array.py
'''
1   2   6   9
l   r   
    l   r
'''

class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        nums.sort()
        l, max_window = 0, 0

        for r in range(len(nums)):
            while nums[l] * k < nums[r]:
                l += 1
            max_window = max(max_window, r - l + 1)

        return len(nums) - max_window

File: 3958-minimum-removals-to-balance-array/minimum-removals-to-balance-array.java
class Solution {
    public int minRemoval(int[] nums, int k) {
        Arrays.sort(nums);
        int left = 0;
        for (int num : nums) {
            if (1L * nums[left] * k < num) {
                left++;
            }
        }
        return left;
    }
}